configure.ac

```diff
@@ -54,7 +65,7 @@ AC_ARG_WITH([mpi],
             [AS_IF([test -d $withval/lib64], [mpi_libdir="lib64"], [mpi_libdir="lib"])
              CFLAGS="-I$withval/include $CFLAGS"
              LDFLAGS="-L$withval/$mpi_libdir $LDFLAGS"],
-            [])
+            [AS_IF([test "$BYPASS_MPICC_CHECK" == "yes"], [], AC_MSG_ERROR([--with-mpi= argument missing]))])
```

Comment:
I don't think you want to ignore the custom path if `mpicc` is found. A user might intentionally want to use a custom MPI installation and this change would no longer allow that. Here's what I was suggesting in the other PR:

    if (--enable-tests) { /* default case */
        if (--with-mpi) {
            Use custom MPI provided and setup LIBS & C/LDFLAGS
        } else {
            See if MPI is in the system path and use it if available / fail if not
        }
    } else {
        /* User explicitly disabled tests, so do nothing */
    }

Even better if you can cut over to using AC_SEARCH_LIBS/pkg-config or the like for the MPI library search.